### PR TITLE
Add a dev guide section for using IntelliJ IDE

### PR DESCRIPTION
**Pull request by devguyio**
### Type of change

- Documentation

### Description

Adds a new IDE section for new contributors using IntelliJ which brings code generation to their attention and instructs them on how to generate the sources in IDE to resolve the "Cannot resolve symbol" errors.

### Checklist

- [x] Update documentation
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: development-docs/DEV_GUIDE.md [hunk 1]
```diff
@@ -91,6 +91,28 @@ before running the `make` commands.
...
+
+Follow these steps to import the project in IntelliJ IDEA, run code generation and trigger a reindex in IntelliJ.
+
+1. After cloning the repo, open the folder in IntelliJ IDEA.
+2. From the toolbar in the [Maven tool window](https://www.jetbrains.com/help/idea/maven-projects-tool-window.html#toolbar), click on `Generate Sources and Update Folders For All Projects` button to initiate the code generation.
+3. Restart the IDE via **Find Action**: press `Ctrl+Shift+A` and type **Restart IDE**.
+
+Afterwards IntelliJ should no longer have any`Cannot resolve symbol` errors. 
+
+## IDE build problems
```
Comment (scholzj) [inline development-docs/DEV_GUIDE.md:106]: Should this be folded under `Using an IDE`? (i.e. have `###` instead of `##`)

**Review by scholzj — APPROVED** [review]

**Review by tombentley — APPROVED** [review]
LGTM. Thanks!

**Review by ppatierno — APPROVED** [review]
LGTM!

File: development-docs/DEV_GUIDE.md [hunk 2]
```diff
@@ -91,6 +91,28 @@ before running the `make` commands.
...
+### IntelliJ IDEA
+The project includes a lot of code generation which takes place within the maven build process, that is why just importing the project might end up with a lot of `Cannot resolve symbol` errors.
+
+Follow these steps to import the project in IntelliJ IDEA, run code generation and trigger a reindex in IntelliJ.
+
+1. After cloning the repo, open the folder in IntelliJ IDEA.
+2. From the toolbar in the [Maven tool window](https://www.jetbrains.com/help/idea/maven-projects-tool-window.html#toolbar), click on `Generate Sources and Update Folders For All Projects` button to initiate the code generation.
+3. Restart the IDE via **Find Action**: press `Ctrl+Shift+A` and type **Restart IDE**.
+
+Afterwards IntelliJ should no longer have any`Cannot resolve symbol` errors.
```
Comment (katheris) [inline development-docs/DEV_GUIDE.md:104]:
```suggestion
Afterwards IntelliJ should no longer have any `Cannot resolve symbol` errors. 
```

**Review by katheris — APPROVED** [review]
Generally looks good to me, only one small comment. The only other suggestion we could make for IntelliJ is I often find after running the Maven build I need to right click on the pom.xml and select Maven > Reload project

**Comment by devguyio** [comment]
Applied suggestions. Thanks!

**Pull request merged**